Tests/test_PDBList.py

```diff
@@ -134,13 +143,89 @@ def test_retrieve_pdb_file_mmtf(self):
         structure = "127d"
         self.check(structure, os.path.join(structure[1:3], f"{structure}.mmtf"), "mmtf")
 
-    def test_double_retrieve(self):
+    def test_double_retrieve_structure(self):
         """Tests retrieving the same file to different directories."""
         structure = "127d"
         self.check(structure, os.path.join("a", f"{structure}.cif"), "mmCif", pdir="a")
         self.check(structure, os.path.join("b", f"{structure}.cif"), "mmCif", pdir="b")
 
 
+class TestPDBListGetAssembly(unittest.TestCase):
+    """Test methods responsible for getting assemblies."""
+
+    @contextlib.contextmanager
+    def make_temp_directory(self, directory):
+        temp_dir = tempfile.mkdtemp(dir=directory)
+        try:
+            yield temp_dir
+        finally:
+            shutil.rmtree(temp_dir)
+
+    def check(self, structure, assembly_num, filename, file_format, pdir=None):
+        with self.make_temp_directory(os.getcwd()) as tmp:
+            pdblist = PDBList(pdb=tmp, obsolete_pdb=os.path.join(tmp, "obsolete"))
+            path = os.path.join(tmp, filename)
+            if pdir:
+                pdir = os.path.join(tmp, pdir)
+            pdblist.retrieve_assembly_file(
+                structure, assembly_num, pdir=pdir, file_format=file_format
+            )
+            self.assertTrue(os.path.isfile(path))
+            os.remove(path)
```

Comment:
You don't need to remove the file explicitly, the temp directory will get removed exiting the `make_temp_directory` decorator

Reply:
Great! I've removed that line in my TestPDBListGetAssembly tests as well as the existing TestPDBListGetStructure tests.